ui/root/src/commonMain/kotlin/social/androiddev/root/root/RootComponentViewModel.kt

```diff
@@ -17,31 +17,23 @@ import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.SupervisorJob
 import kotlinx.coroutines.cancel
 import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.launch
 import social.androiddev.domain.authentication.model.AuthStatus
 import social.androiddev.domain.authentication.usecase.GetAuthStatus
 import kotlin.coroutines.CoroutineContext
 
-interface RootComponentViewModel : InstanceKeeper.Instance {
-    val authState: StateFlow<AuthStatus?>
-}
-
-fun RootComponentViewModel(coroutineContext: CoroutineContext, getAuthStatus: GetAuthStatus): RootComponentViewModel =
-    RealRootComponentViewModel(coroutineContext, getAuthStatus)
-
-private class RealRootComponentViewModel(coroutineContext: CoroutineContext, getAuthStatus: GetAuthStatus) :
-    RootComponentViewModel {
+class RootComponentViewModel(coroutineContext: CoroutineContext, getAuthStatus: GetAuthStatus) :
+    InstanceKeeper.Instance {
     private val viewModelScope = CoroutineScope(coroutineContext + SupervisorJob())
 
-    private val _authState = MutableStateFlow<AuthStatus?>(null)
-    override val authState = _authState.asStateFlow()
+    private val _authState = MutableStateFlow<RootComponent.UiAuthStatus>(RootComponent.UiAuthStatus.Unknown)
```

Comment:
You can still simplify the `init` block and consolidate this private and public property using my suggestion from your other PR. Here's the link to my comment
https://github.com/AndroidDev-social/DodoForMastodon/pull/111#discussion_r1061676320

Reply:
the `stateIn` call is suspend call, so can't do it that way

Reply:
I'm not sure I'm following... Please see the doc
https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/state-in.html

Reply:
Your repo returns a Flow.
You should be able to do something similar to 
```kotlin
val state: StateFlow<State> = repo.getFlow().stateIn(scope, SharingStarted.Eagerly, State.LOADING)
```

Reply:
That'd be true, if `stateIn` would not be a suspend call - The docs mention the function signature (at the bottom).

> Starts the upstream flow in a given scope, suspends until the first value is emitted, and returns a hotStateFlow of future emissions

The function signature is 
```kotlin
suspend fun <T> Flow<T>.stateIn(scope: CoroutineScope): StateFlow<T>
```

Reply:
Ah, perfect - That'd work!